site/app/controllers/admin/PlagiarismController.php

```diff
@@ -464,96 +521,105 @@ public function saveNewPlagiarismConfiguration($new_or_edit, $gradeable_id = nul
         }
 
 
-        // Save the Lichen run timestamp file
-        $current_time = $this->core->getDateTimeNow()->format("Y-m-d H:i:sO");
-        $current_time_string_tz = $current_time . " " . $this->core->getConfig()->getTimezone()->getName();
-        if (!@file_put_contents(FileUtils::joinPaths($course_path, "lichen", "config", "." . $gradeable_id . ".lichenrun.timestamp"), $current_time_string_tz . "\n")) {
-            $this->core->addErrorMessage("Failed to save timestamp file for this Lichen Run. Create the configuration again.");
-            $this->core->redirect($return_url);
-        }
-
-
-        // Create the Lichen job
-        $ret = $this->enqueueLichenJob("RunLichen", $gradeable_id);
+        // Create the Lichen job ///////////////////////////////////////////////
+        $ret = $this->enqueueLichenJob("RunLichen", $gradeable_id, $config_id);
         if ($ret !== null) {
             $this->core->addErrorMessage("Failed to add configuration to Lichen queue. Create the configuration again.");
             $this->core->redirect($return_url);
         }
 
-        $this->core->addSuccessMessage("Lichen Plagiarism Detection configuration created for " . $gradeable_id);
+        $this->core->addSuccessMessage("Lichen Plagiarism Detection configuration created for {$gradeable_id} configuration #{$config_id}");
         $this->core->redirect($this->core->buildCourseUrl(['plagiarism']) . '?' . http_build_query(['refresh_page' => 'REFRESH_ME']));
     }
 
-    private function enqueueLichenJob($job, $gradeable_id) {
+
+    /**
+     * @param string $job
+     * @param string $gradeable_id
+     * @param string $config_id
+     * @return string|null
+     */
+    private function enqueueLichenJob(string $job, string $gradeable_id, string $config_id) {
         $semester = $this->core->getConfig()->getSemester();
         $course = $this->core->getConfig()->getCourse();
 
         $lichen_job_data = [
             "job" => $job,
             "semester" => $semester,
             "course" => $course,
-            "gradeable" => $gradeable_id
+            "gradeable" => $gradeable_id,
+            "config_id" => $config_id
         ];
-        $lichen_job_file = "/var/local/submitty/daemon_job_queue/lichen__" . $semester . "__" . $course . "__" . $gradeable_id . ".json";
+
+        try {
+            $this->verifyGradeableAndConfigAreValid($gradeable_id, $config_id);
+        }
+        catch (\Exception $e) {
+            return $e;
```

Comment:
Do you want to return the whole exception in a function you said you will return a string or null?